RT/Renderer/Backend/DX12/src/ImageReadWrite.cpp

```diff
@@ -62,112 +126,195 @@ void RT_WritePNGToDisk(const char *path, int w, int h, int channel_count, const
 	RT_ArenaMemoryScope(&g_thread_arena)
 	{
 		int result = stbi_write_png(path, w, h, channel_count, pixels, stride_in_bytes);
-		RT_ASSERT(result); // TODO: Cool error handling / return an error code at least
+
+		if (!result)
+		{
+			fprintf(stderr, "[RT_WritePNGToDisk]: ERROR: Failed to write png: '%s'\n", path);
+		}
 	}
 }
 
-//  Code modified from DirectXTK12 : LoaderHelpers::LoadTextureDataFromFile (https://github.com/microsoft/DirectXTK12/blob/main/Src/LoaderHelpers.h)
-bool RT_LoadDDSImageFromDisk(RT_Arena* arena, const char* path, const uint8_t** ddsData, const struct DDS_HEADER** header, const uint8_t** bitData, size_t* bitSize)
+RT_Image RT_LoadDDSFromMemory(RT_String memory)
 {
-	RT_ArenaMarker marker = RT_ArenaGetMarker(&g_thread_arena);
+	RT_Image result = {};
 
-	const uint32_t DDS_MAGIC = 0x20534444; // "DDS "
+	if (memory.count > 4294967296)
+	{
+		fprintf(stderr, "[RT_LoadDDSFromMemory]: ERROR: File too large, what are you doing! Your texture is more than 4GB!?\n");
+		return result;
+	}
 
-	FILE* f;
+	if (memory.count < sizeof(uint32_t) + sizeof(DDS_HEADER))
+	{
+		fprintf(stderr, "[RT_LoadDDSFromMemory]: ERROR: File too small to fit a valid dds texture.\n");
+		return result;
+	}
+
+	char *data = memory.bytes;
 
-	if (0 == fopen_s(&f, path, "rb"))
+	const uint32_t dds_magic = 0x20534444; // "DDS "
+
+	if (memcmp(data, &dds_magic, 4) != 0)
 	{
-		long fileSize = 0;
-		
-		fseek(f, 0, SEEK_END); // seek to end of file
-		fileSize = ftell(f); // get current file pointer
-		fseek(f, 0, SEEK_SET); // seek back to beginning of file
+		fprintf(stderr, "RT_LoadDDSFromMemory]: ERROR: File did not start with \"DDS \", which does not match my expectations...\n");
```

Comment:
Missing opening square bracket in the printf….

Reply:
*smh* I expected more of you